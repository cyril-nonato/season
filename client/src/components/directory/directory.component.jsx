import React from 'react'
import DirectoryItem from '../directory-item/directory-item.component';

import * as S from './directory.styles';

const Directory = ({sections}) => {
  return ( 
    <S.Container>
      {
        sections.map(section => <DirectoryItem key={section.id} item={section} />)
      }
    </S.Container>
   );
}
 
export default Directory;