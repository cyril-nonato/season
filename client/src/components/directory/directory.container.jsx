import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import { selectSections } from '../../redux/home/home.selector';
import Directory from './directory.component';

const mapStateToProps = createStructuredSelector({
  sections: selectSections
});

const DirectoryContainer = compose(connect(mapStateToProps))(Directory);

export default DirectoryContainer;