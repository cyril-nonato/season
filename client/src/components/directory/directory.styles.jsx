import styled from 'styled-components';
import { mediaQueries } from '../../theme/theme';

export const Container = styled.div`
  width: 100%;
  margin: 0 auto;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-gap: 2rem;
  margin-bottom: 5rem;

  @media only screen and (max-width: ${mediaQueries.windows}) {
    grid-template-columns: 1fr 1fr 1fr;
  };

  @media only screen and (max-width: ${mediaQueries.tablet}) {
    grid-template-columns: 1fr 1fr;
  };

  @media only screen and (max-width: ${mediaQueries.phone}) {
    grid-template-columns: 1fr;
  }
`;