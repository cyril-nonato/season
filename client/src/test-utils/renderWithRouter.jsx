import React from 'react'
import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom'
import {render} from '@testing-library/react'

export default function renderWithRouter(
  component,
  {
    route = '/',
    history = createMemoryHistory({ initialEntries: [route] }),
  } = {}
) {
  const Wrapper = ({ children }) => (
    <Router history={history}>{children}</Router>
  )

  return {
    ...render(component, { wrapper: Wrapper }),
    history
  }
}