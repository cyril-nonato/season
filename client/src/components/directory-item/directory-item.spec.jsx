import React from 'react'
import { cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import DirectoryItem from './directory-item.component';
import renderWithRouter from '../../test-utils/renderWithRouter';
import 'jest-styled-components'
import * as S from './directory.styles';

afterEach(cleanup);

describe('Directory Item test', () => {
  const item = {
    logo: 'some-logo',
    name: 'some-name',
    linkUrl: 'url'
  }

  test('Should render the props correctly', () => {
    const { getByTestId, asFragment } = renderWithRouter(<DirectoryItem item={item} />)
    expect(getByTestId('link')).toHaveAttribute('href', `/${item.linkUrl}`);
    expect(getByTestId('image')).toHaveAttribute('src', item.logo);
    expect(getByTestId('h3')).toHaveTextContent(item.name);
  });

  test('Should match the snapshot', () => {
    const { asFragment } = renderWithRouter(<DirectoryItem item={item} />)

    expect(asFragment()).toMatchSnapshot();
  })
});
