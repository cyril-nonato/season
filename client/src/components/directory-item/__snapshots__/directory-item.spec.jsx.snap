// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory Item test Should match the snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-bottom: 1rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 5rem;
  text-transform: uppercase;
  font-size: 1.8rem;
  padding: 1rem 0;
  -webkit-transition: all .3s ease-in-out;
  transition: all .3s ease-in-out;
}

.c0:hover {
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.c0:active {
  -webkit-transform: translateY(0);
  -ms-transform: translateY(0);
  transform: translateY(0);
}

.c0:link,
.c0:visited {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: black;
}

.c1 {
  display: block;
  height: 100px;
  width: 100px;
}

.c2 {
  margin-top: 2rem;
}

@media screen and (min-width:700px) {
  .c1 {
    height: 150px;
    width: 150px;
  }
}

<a
    class="c0"
    data-testid="link"
    href="/url"
  >
    <img
      alt="some-name"
      class="c1"
      data-testid="image"
      src="some-logo"
    />
    <h3
      class="c2"
      data-testid="h3"
    >
      some-name
    </h3>
  </a>
</DocumentFragment>
`;
